CUDA.Dockerfile

```diff
@@ -0,0 +1,75 @@
+# Build argumnets
+ARG CUDA_VER=11.7
+ARG UBUNTU_VER=22.04
+
+# Download the base image
+FROM nvidia/cuda:${CUDA_VER}.0-devel-ubuntu${UBUNTU_VER}
+# you can check for all available images at https://hub.docker.com/r/nvidia/cuda/tags
+
+# Install as root
+USER root
+
+# Install dependencies
+RUN apt-get update
+RUN apt-get upgrade -y
+RUN apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+    
+RUN apt-get update && \
+    DEBIAN_FRONTEND="noninteractive" apt-get install --yes \
+    --no-install-recommends \
+    bash \
+    bash-completion \
+    cmake \
+    curl \
+    git \
+    libopenblas-dev \
+    linux-headers-$(uname -r) \
+    nano \
+    python3 python3-dev python3-pip python-is-python3 \
+    sudo \
+    wget
+
+RUN apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# Add a user `${USERNAME}` so that you're not developing as the `root` user
+ENV USERNAME=ibm
+RUN echo "${USERNAME}"
+RUN useradd ${USERNAME} \
+    --create-home \
+    --shell=/bin/bash \
+    --user-group && \
+    echo "${USERNAME} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/nopasswd
+
+# Change to your user
+USER ${USERNAME}
+WORKDIR /home/${USERNAME}
+
+# Install python packages as your user
+RUN pip install --upgrade pip
+RUN pip install pybind11 scikit-build
+RUN pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu116
+
+# Set path of python packages
+RUN echo 'export PATH=$HOME/.local/bin:$PATH' >> /home/${USERNAME}/.bashrc
+
+# clone the repo and change to the source directory
+RUN git clone https://github.com/IBM/aihwkit.git
+WORKDIR /home/${USERNAME}/aihwkit
+
+# Default value for NVIDIA RTX A5000, find your own GPU model and replace it
+# use the k: https://developer.nvidia.com/cuda-gpus
+ARG CUDA_ARCH=86
+ARG NUM_CPU_THREADS=8
+RUN echo "Detected CUDA_ARCHITECTURE is = ${CUDA_ARCH}"
+# Build and install IBM aihwkit
+RUN make clean
+RUN python setup.py build_ext -j${NUM_CPU_THREADS} --inplace -DUSE_CUDA=ON -DRPU_CUDA_ARCHITECTURES="${CUDA_ARCH}" -DRPU_BLAS=OpenBLAS
+
+# Set paths of installation
+RUN echo 'export PYTHONPATH=$HOME/aihwkit/src' >> /home/${USERNAME}/.bashrc
```

Comment:
Can you replace these two lines with installing `aihwkit` via `pip`, rather than compiling the shared library and manually apppending it to the path? This would allow for the package to be available same as other python packages in a cleaner way.

Hopefully this (adapted) bit from the [docs](https://aihwkit.readthedocs.io/en/latest/advanced_install.html) would do the trick for passing cmake options to `pip install`:
```
pip install . --install-option="-DUSE_CUDA=ON" ...

Reply:
(On a more minor note, while editing the above, if you could get rid of the empty newlines at the end of the file would be great!)

Reply:
This is resolved.

Reply:
Thanks!